gradient_estimators/random_gradient_estimator.py

```diff
@@ -220,10 +240,20 @@ def update_model_given_seed_and_grad(
         iteration_grad_scalar: Sequence[torch.Tensor],
     ) -> None:
         assert len(iteration_seeds) == len(iteration_grad_scalar)
+
+        if self.sgd_only_no_optim:
+            lr = optimizer.defaults["lr"]  # Assume only one parameter group with lr.
+            assert self.paramwise_perturb
+            for one_update_seed, one_update_grad_dirs in zip(
+                iteration_seeds, iteration_grad_scalar
+            ):
+                self.sgd_no_optim_update_model(one_update_grad_dirs, one_update_seed, lr)
+            return
+
         # NOTE: this zero_grad operation is critical since it sets the parameter.grad to None
         # which is checked in self.generate_then_put_grad_paramwise
```

Comment:
Feel the comment seems not relevant anymore?

we assign value when i == 0, this should decouple 2 parts of logic.

Reply:
like even we do not zero_grad, this code should still work

Reply:
right, delete that comment